cloudigrade/account/tasks.py

```diff
@@ -44,60 +44,83 @@ def copy_ami_snapshot(arn, ami_id, source_region):
     """
     session = aws.get_session(arn)
     ami = aws.get_ami(session, ami_id, source_region)
-    snapshot_id = aws.get_ami_snapshot_id(ami)
-    snapshot = aws.get_snapshot(session, snapshot_id, source_region)
+    customer_snapshot_id = aws.get_ami_snapshot_id(ami)
+    customer_snapshot = aws.get_snapshot(
+        session, customer_snapshot_id, source_region)
 
-    if snapshot.encrypted:
+    if customer_snapshot.encrypted:
         image = AwsMachineImage.objects.get(ec2_ami_id=ami_id)
         image.is_encrypted = True
         image.save()
         raise AwsSnapshotEncryptedError
 
-    aws.add_snapshot_ownership(snapshot)
+    aws.add_snapshot_ownership(customer_snapshot)
 
-    new_snapshot_id = aws.copy_snapshot(snapshot_id, source_region)
-    create_volume.delay(ami_id, new_snapshot_id)
+    snapshot_copy_id = aws.copy_snapshot(customer_snapshot_id, source_region)
+    volume_information = {
+        'arn': arn,
+        'source_region': source_region,
+        'ami_id': ami_id,
+        'customer_snapshot_id': customer_snapshot_id,
+        'snapshot_copy_id': snapshot_copy_id
+    }
+    logger.info(_(
+        '{0}: customer_snapshot_id={1}, snapshot_copy_id={2}').format(
+        'copy_ami_snapshot',
+        volume_information['customer_snapshot_id'],
+        volume_information['snapshot_copy_id']))
+    create_volume.delay(volume_information)
 
 
 @retriable_shared_task
 @rewrap_aws_errors
-def create_volume(ami_id, snapshot_id):
+def create_volume(volume_information):
     """
     Create an AWS Volume in the primary AWS account.
 
     Args:
-        ami_id (str): The AWS AMI id for which this request originated
-        snapshot_id (str): The id of the snapshot to use for the volume
+        volume_information (dict): Meta-data about scanned instances (arn,
+            source_region, ami_id, customer_snapshot_id, snapshot_copy_id)
 
     Returns:
         None: Run as an asynchronous Celery task.
 
     """
     zone = settings.HOUNDIGRADE_AWS_AVAILABILITY_ZONE
-    volume_id = aws.create_volume(snapshot_id, zone)
+    volume_id = aws.create_volume(volume_information['snapshot_copy_id'], zone)
```

Comment:
Part of `create_volume()` is to verify that the snapshot is ready, next line would be a great place to fire off a new task with a `.delay()` that would remove ownership from from the customers account snapshot since we know that our copy was finished.